js-library/src/utils/keys.ts

```diff
@@ -0,0 +1,41 @@
+import { utils } from 'ethers'
+import { isHexString } from 'ethers/lib/utils'
+import {
+  HexString,
+  PublicKey,
+  PublicKeyX,
+  PublicKeyY,
+  PUBLIC_KEY_LENGTH,
+  PUBLIC_KEY_PART_LENGTH,
+} from '../model/hex.types'
+
+const { hexStripZeros } = utils
+
+export function isHexStringPublicKey(hexString: HexString): boolean {
+  return isHexString(hexString) && hexString.length === PUBLIC_KEY_LENGTH
+}
+
+export function isHexStringPublicKeyPart(hexString: HexString): boolean {
+  return isHexString(hexString) && hexString.length === PUBLIC_KEY_PART_LENGTH
+}
+
+export function isPublicKeyValid(publicKey: PublicKey): boolean {
+  return isHexStringPublicKey(publicKey) && hexStripZeros('0x' + publicKey.substring(4)) !== '0x'
+}
+
+export function splitPublicKey(publicKey: PublicKey): [PublicKeyX, PublicKeyY] {
+  if (!isPublicKeyValid(publicKey)) {
+    throw new Error('Public key is not valid.')
+  }
```

Comment:
you can place it into an [assertion function](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions) (that uses the before mentioned type-guard) and the code seems clearer